MarketplaceService/Repositories/DAppRepository.cs

```diff
@@ -46,6 +49,27 @@ public async Task DeleteDAppOffer(Guid id)
             await _dAppOffers.DeleteManyAsync(f => f.Id == id);
         }
 
+        public async Task<PagedList<DAppOffer>> GetAllDAppOffers(GetDAppOfferModel getDAppOfferModel)
+        {
```

Comment:
This query is too heavy make it one query which does it all because now you get all objects.